frontend/src/metabase/dashboard/components/DashboardHeader.jsx

```diff
@@ -106,6 +108,15 @@ export default class DashboardHeader extends Component {
         this.props.onEditingChange(true);
     }
 
+    async onAddTextBox() {
```

Comment:
I'm guessing this logic should be moved somewhere else. Possibly `dashboard.js`, thoughts?

Reply:
yup, our usual idiom is to wrap those in redux thunkActions.

Reply:
How about now that I'm not doing any async behavior? Should this logic be moved? If so, where?

Reply:
yeah this could be in `dashboard.js` – it doesn't really change the situation if there is async behavior or not! You seem to already have added `addDashCardToDashboard` and you can just follow the same pattern for this (maybe `addTextDashCardToDashboard` or something like that).